File: Backend/src/main/java/Circuit_De_Stage/Backend/Entities/UserDocumentSeen.java
package Circuit_De_Stage.Backend.Entities;


import Circuit_De_Stage.Backend.Entities.Enum.RoleType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "UserDocumentSeen")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDocumentSeen {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private boolean seen= false;

    @Enumerated(EnumType.STRING)
    private RoleType role; 
    
    
    


    @ManyToOne
    @JoinColumn(name = "document_id")
    private Document document;

    @ManyToOne
    @JoinColumn(name = "utilisateur_id")
    private User utilisateur;

    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public boolean isSeen() {
		return seen;
	}

	public void setSeen(boolean seen) {
		this.seen = seen;
	}

	public RoleType getRole() {
		return role;
	}

	public void setRole(RoleType role) {
		this.role = role;
	}

	public Document getDocument() {
		return document;
	}

	public void setDocument(Document document) {
		this.document = document;
	}

	public User getUtilisateur() {
		return utilisateur;
	}

	public void setUtilisateur(User utilisateur) {
		this.utilisateur = utilisateur;
	}

    
    
    
    
    
}

File: Frontend/src/app/classes/user-document-seen.ts
import { Document } from './document';
import { RoleType } from './enums/role-type';
import { User } from './user';

export class UserDocumentSeen {
    id: number;
    seen: boolean;
    role: RoleType;
    document: Document;
    utilisateur: User;

    constructor(
        id: number,
        seen: boolean,
        role: RoleType,
        document: Document,
        utilisateur: User
    ) {
        this.id = id;
        this.seen = seen;
        this.role = role;
        this.document = document;
        this.utilisateur = utilisateur;
    }
}